diff -Naur libodbc++-0.2.3/include/odbc++/callablestatement.h libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/callablestatement.h
--- libodbc++-0.2.3/include/odbc++/callablestatement.h	2003-03-20 16:52:27.000000000 +0100
+++ libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/callablestatement.h	2005-01-18 14:47:32.761605624 +0100
@@ -160,7 +160,7 @@
   };
 
   
-}; // namespace odbc
+} // namespace odbc
 
 
 #endif // __ODBCXX_CALLABLESTATEMENT_H
diff -Naur libodbc++-0.2.3/include/odbc++/connection.h libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/connection.h
--- libodbc++-0.2.3/include/odbc++/connection.h	2001-06-12 12:38:16.000000000 +0200
+++ libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/connection.h	2005-01-18 14:47:41.210321224 +0100
@@ -204,7 +204,7 @@
 
 
 
-}; // namespace odbc
+} // namespace odbc
 
 
 #endif // __ODBCXX_CONNECTION_H
diff -Naur libodbc++-0.2.3/include/odbc++/databasemetadata.h libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/databasemetadata.h
--- libodbc++-0.2.3/include/odbc++/databasemetadata.h	2001-06-12 12:38:17.000000000 +0200
+++ libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/databasemetadata.h	2005-01-18 14:47:44.922756848 +0100
@@ -1333,7 +1333,7 @@
   };
 
   
-}; // namespace odbc
+} // namespace odbc
 
 
 #endif // __ODBCXX_DATABASEMETADATA_H
diff -Naur libodbc++-0.2.3/include/odbc++/drivermanager.h libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/drivermanager.h
--- libodbc++-0.2.3/include/odbc++/drivermanager.h	2001-06-12 12:38:17.000000000 +0200
+++ libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/drivermanager.h	2005-01-18 14:47:48.704181984 +0100
@@ -147,7 +147,7 @@
 
 
 
-};
+}
 
 
 #endif // __ODBCXX_DRIVERMANAGER_H
diff -Naur libodbc++-0.2.3/include/odbc++/errorhandler.h libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/errorhandler.h
--- libodbc++-0.2.3/include/odbc++/errorhandler.h	2001-06-12 12:38:18.000000000 +0200
+++ libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/errorhandler.h	2005-01-18 14:47:53.838401464 +0100
@@ -127,6 +127,6 @@
   };
 
   
-}; // namespace odbc
+} // namespace odbc
 
 #endif
diff -Naur libodbc++-0.2.3/include/odbc++/preparedstatement.h libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/preparedstatement.h
--- libodbc++-0.2.3/include/odbc++/preparedstatement.h	2003-05-09 10:45:48.000000000 +0200
+++ libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/preparedstatement.h	2005-01-18 14:47:57.663819912 +0100
@@ -211,6 +211,6 @@
   };
 
 
-}; // namespace odbc
+} // namespace odbc
 
 #endif // __ODBCXX_PREPAREDSTATEMENT_H
diff -Naur libodbc++-0.2.3/include/odbc++/resultset.h libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/resultset.h
--- libodbc++-0.2.3/include/odbc++/resultset.h	2001-06-12 12:38:19.000000000 +0200
+++ libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/resultset.h	2005-01-18 14:48:01.966165856 +0100
@@ -619,7 +619,7 @@
 
   
 
-}; // namespace odbc
+} // namespace odbc
 
 
 #endif // __ODBCXX_RESULTSET_H
diff -Naur libodbc++-0.2.3/include/odbc++/resultsetmetadata.h libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/resultsetmetadata.h
--- libodbc++-0.2.3/include/odbc++/resultsetmetadata.h	2001-06-12 12:38:20.000000000 +0200
+++ libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/resultsetmetadata.h	2005-01-18 14:48:05.372647992 +0100
@@ -176,7 +176,7 @@
 
   
 
-}; // namespace odbc
+} // namespace odbc
 
 
 #endif // __ODBCXX_RESULTSETMETADATA_H
diff -Naur libodbc++-0.2.3/include/odbc++/statement.h libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/statement.h
--- libodbc++-0.2.3/include/odbc++/statement.h	2003-03-20 16:52:32.000000000 +0100
+++ libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/statement.h	2005-01-18 14:48:11.589702856 +0100
@@ -260,7 +260,7 @@
   
 
 
-}; // namespace odbc
+} // namespace odbc
 
 
 #endif // __ODBCXX_STATEMENT_H
diff -Naur libodbc++-0.2.3/include/odbc++/threads.h libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/threads.h
--- libodbc++-0.2.3/include/odbc++/threads.h	2001-06-12 12:38:20.000000000 +0200
+++ libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/threads.h	2005-01-18 14:48:16.409970064 +0100
@@ -65,7 +65,7 @@
     }
   };
 
-}; //namespace odbc
+} //namespace odbc
 
 // macro used all over the place
 #define ODBCXX_LOCKER(mut) odbc::Locker _locker(mut)
diff -Naur libodbc++-0.2.3/include/odbc++/types.h libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/types.h
--- libodbc++-0.2.3/include/odbc++/types.h	2003-06-17 12:08:02.000000000 +0200
+++ libodbc++-0.2.3-no-namespace-closing-colon/include/odbc++/types.h	2005-01-18 14:48:22.560035112 +0100
@@ -791,7 +791,7 @@
     }
   };
 
-}; // namespace odbc
+} // namespace odbc
 
 
 #endif // __ODBCXX_TYPES_H
