--- livesupport-1.0/etc/Makefile.in	2005-08-25 13:09:41.000000000 +0200
+++ livesupport-1.0-gentoo/etc/Makefile.in	2005-09-21 13:26:08.000000000 +0200
@@ -54,7 +54,10 @@
 
 USR_DIR     = ${prefix}
 USR_BIN_DIR = ${USR_DIR}/bin
+USR_DOC_DIR = ${USR_DIR}/doc
 USR_ETC_DIR = ${USR_DIR}/etc
+USR_LIB_DIR = ${USR_DIR}/lib
+USR_VAR_DIR = ${USR_DIR}/var
 
 HOSTNAME          		= @HOSTNAME@
 APACHE_GROUP      		= @APACHE_GROUP@
@@ -426,9 +430,10 @@
 #-------------------------------------------------------------------------------
 #   Installation related targets
 #-------------------------------------------------------------------------------
-.PHONY: install create_database install_modules install_products
+.PHONY: install create_database setup_install_dirs
+.PHONY: install_modules install_products
 
-install: setup compile create_database install_modules install_products
+install: setup compile setup_install_dirs create_database install_modules install_products
 	${MKDIR} ${USR_ETC_DIR}/apache
 	${CP} ${ETC_DIR}/apache/*.conf ${USR_ETC_DIR}/apache
 	${MKDIR} ${USR_BIN_DIR}
@@ -444,6 +449,13 @@
                                            --dbpassword=${DB_PASSWORD}
 endif
 
+setup_install_dirs:
+	${MKDIR} ${USR_BIN_DIR}
+	${MKDIR} ${USR_DOC_DIR}
+	${MKDIR} ${USR_ETC_DIR}
+	${MKDIR} ${USR_LIB_DIR}
+	${MKDIR} ${USR_VAR_DIR}
+
 install_modules:
 	${MAKE} -C ${ALIB_DIR} install
 	${MAKE} -C ${ARCHIVE_SERVER_DIR} install
