Index: /usr/lib/airtime/airtime_virtualenv/lib/python2.6/site-packages/mutagen/mp3.py
===================================================================
--- /usr/lib/airtime/airtime_virtualenv/lib/python2.6/site-packages/mutagen/mp3.py	(revision 108)
+++ /usr/lib/airtime/airtime_virtualenv/lib/python2.6/site-packages/mutagen/mp3.py	(working copy)
@@ -151,15 +151,16 @@
         self.sample_rate = self.__RATES[self.version][sample_rate]
 
         if self.layer == 1:
-            frame_length = (12 * self.bitrate / self.sample_rate + padding) * 4
+            padding *= 4
             frame_size = 384
         elif self.version >= 2 and self.layer == 3:
-            frame_length = 72 * self.bitrate / self.sample_rate + padding
             frame_size = 576
         else:
-            frame_length = 144 * self.bitrate / self.sample_rate + padding
             frame_size = 1152
 
+        frame_length = frame_size / 8 * \
+            self.bitrate / self.sample_rate + padding
+
         if check_second:
             possible = frame_1 + frame_length
             if possible > len(data) + 4:
@@ -172,9 +173,7 @@
             if frame_data & 0xFFE0 != 0xFFE0:
                 raise HeaderNotFoundError("can't sync to second MPEG frame")
 
-        frame_count = real_size / float(frame_length)
-        samples = frame_size * frame_count
-        self.length = samples / self.sample_rate
+        self.length = 8 * real_size / float(self.bitrate)
 
         # Try to find/parse the Xing header, which trumps the above length
         # and bitrate calculation.
