--- gst-plugins-0.8.12/gst/typefind/gsttypefindfunctions.c	2006-01-29 23:54:47.000000000 +0100
+++ gst-plugins-0.8.12-livesupport/gst/typefind/gsttypefindfunctions.c	2006-10-03 12:17:53.000000000 +0200
@@ -143,7 +143,7 @@
 
 /*** application/xml **********************************************************/
 
-#define XML_BUFFER_SIZE 256
+#define XML_BUFFER_SIZE 16     /* If the XML file is < 16 bytes we're screwed */
 #define XML_INC_BUFFER {						\
   pos++;								\
   if (pos == XML_BUFFER_SIZE) {						\
